README.md

```diff
@@ -162,9 +180,86 @@ specifies channel 0 which is connected to the thermometer.
     iex> volts = counts / 1023 * 3.3
     1.461290322580645
 
+Since many people coming to `elixir_ale` may be new to reading data sheets 
+and working with bitstrings, the following is a more detailed explanation.
+
+First we call SPI.start_link with Linux spidev device driver as it argument.  
+We pattern match with `{:ok, pid}` so we can use `pid` in the 
+SPI.transfer function next.
+
+    {:ok, pid} = SPI.start_link("spidev0.0")
+
+`elixir_ale` uses the Linux spidev device driver for communicating with SPI 
+devices. It creates files in /dev for each SPI interface that it knows about. 
+Many boards only have one SPI interface and it shows up as /dev/spidev0.0. 
+The easiest way to find out the right name is to run `ls spidev*` in the /dev 
+directory. If nothing shows up, it is likely that the spidev driver needs to 
+be loaded. Try running modprobe spidev or consult your board's documentation.
+
+Next we'll make a call to SPI.transfer to read the potentiometer.
+
+    SPI.transfer(pid, <<0x70, 0x00>>)
+
+As mentioned above, the protocol for talking to the ADC is described in the 
+[MCP3002](http://www.microchip.com/wwwproducts/en/MCP3002) data sheet. See 
+Figure 6-1, the request protocol is labelled "MCU Transmitted Data." The 
+leftmost bit is the "Start" bit. The second bit is SGL/DIFF, the third bit is 
+ODD/SIGN, and the fourth bit is MS BF. You may notice a `x` before the start 
+bit which represents a leading `0`.  This is often done when dealing with 
+microcontroller-based SPI ports that must send 8 bits at a time. The request 
+and response are happening simultaneously so the size of the request (in bits) 
+must equal the size of the response. In this case, the request consists of 4 
+bits and the response contains 11 bits (a null bit and 10 data bits) for a 
+total of 15 bits, so a leading `0` is necessary to get 16 bits. The ADC doesn't 
+care about the 11 bits at the end of the request so we can just send 11 `0` bits.
+
+Table 5-1 explains how to configure the request bits. For example, SGL/DIFF==1 
+and ODD/SIGN==1 specifies single-ended mode from channel 1 which has been 
+connected to the potentiometer. MSBF==0 will output the result of the conversion 
+with MSB (Most Significant Bit) first. TODO verify MSBF info
```

Comment:
> Before it was 0x74 and 0x64. Then I changed it to 0x70 and 0x60. MSBF is the 0x08 bit, so it should have been low both times, right?
> The 0x04 bit should be in the don't care section, so that's why I got rid of it. Let me read the MSBF section. Maybe we do want it...
> Yeah, I think that we want it on, but I see why it doesn't matter. When it's off, it means that the MCP3002 outputs the value MSB first (the way we want), and then if we keep asking for data, we'll get the LSB version. If MSBF is on, then we couldn't keep clocking out data.
> We don't ask for more data, so we don't get the LSB version. Since CS goes high at the end of our transaction and then low at the beginning of the next, the chip is ok.
> This is a good catch. We should make it 0x78 and 0x68 to be completely correct.